aiida/cmdline/commands/cmd_group.py

```diff
@@ -47,12 +47,30 @@ def group_add_nodes(group, force, nodes):
 @with_dbenv()
 def group_remove_nodes(group, nodes, clear, force):
     """Remove nodes from a group."""
-    if clear:
-        message = f'Do you really want to remove ALL the nodes from Group<{group.label}>?'
-    else:
-        message = f'Do you really want to remove {len(nodes)} nodes from Group<{group.label}>?'
+    from aiida.orm import QueryBuilder, Group, Node
 
     if not force:
+        node_pks = [node.pk for node in nodes]
+
+        query = QueryBuilder()
+        query.append(Group, filters={'id': group.pk}, tag='group')
+        query.append(Node, with_group='group', filters={'id': {'in': node_pks}}, project='id')
+
+        group_node_pks = query.all(flat=True)
```

Comment:
Here I am not 100% sure which way is faster, the one you've implemented or something like:

```python
query = QueryBuilder()
query.append(Group, filters={'id': group.pk}, tag='group').append(Node)
group_nodes = set(query.all(flat=True))

group_nodes = group_nodes.intersect(nodes)
```

Reply:
I discovered an interesting bug while testing this: https://github.com/aiidateam/aiida-core/issues/4736

Reply:
AiiDA nodes are not hashable (in the Python sense of having a `__hash__` method), so you can't put them in a `set`.